File: 0422/비밀지도.py
# 되게 이상하게 짰는데 되기는 되네. 다른 애들꺼 코드 한번 봐야겠다.
def ijin(num, arr, n, i):
    nn = n - 1
    while num > 0:
        if num >= 2**nn:
            num -= 2**nn
            arr[i][n - nn - 1] = '#'
        nn -= 1

def solution(n, arr1, arr2):
    answer = []
    arr_map = [ [0] * n for _ in range(n) ]
    # arr2_map = [ [0] * n for _ in range(n) ]
    for i in range(n):
      ar = arr1[i]
      ijin(ar, arr_map, n, i)
      ar2 = arr2[i]
      ijin(ar2, arr_map, n, i)
    
    # 얘 항상 까묵는다...ㅎ
    for arr in arr_map:
        # answer.append(''.join(map(str, arr))) 일단 버리고
        temp = ''
        for ar in arr:
            if ar == '#':
                temp += '#'
            else:
                temp += ' '
        answer.append(temp)

    return answer

print(solution(5, [9, 20, 28, 18, 11], [30, 1, 21, 17, 28]))

File: 2018_KAKAO_BLIND_RECRUITMENT/비밀지도.js
function solution(n, arr1, arr2) {
    const answer = [];

    for(let i = 0; i < n; i ++){
        let arr = arr1[i] | arr2[i]; // 이거 or 처리를 먼저 하는거네.... 흐음 이거 비트연산자자누...
        console.log('arr1: '+ arr1[i])  // + 로 하면 자동으로 string 변환
        console.log('arr2: ', arr2[i])  // int값은 노란색으로 되네 싱기 ㅇㅅㅇ
        console.log(arr)
        console.log('toString 결과: ', arr.toString(2).replace(/1/g, '#').replace(/0/g, ' '))
        console.log('padStart 결과: ', arr.toString(2).padStart(n, '0').replace(/1/g, '#').replace(/0/g, ' '))
        answer.push(arr.toString(2).padStart(n, '0').replace(/1/g, '#').replace(/0/g, ' ')); // 이거 뭔가 엄청 사기적이자나...
        console.log(answer)
    }

    return answer;
}

console.log(solution(5, [9, 20, 28, 18, 11], [30, 1, 21, 17, 28]))  // 이렇게 해줘도 나옴.